### PR TITLE
Add language selector in the navbar

### PR DESCRIPTION
**Pull request by renny-ren**
Add language selector in the navbar

### CODE REVIEW

File: src/index.js
Comment (lopopolo) [inline src/index.js]: I'd prefer using DOM Events than `window.onload`. Can we replace this with a `DOMContentLoaded` event handler? I'm not sure how that interacts with the js bundle being an async module though.

```suggestion
document.addEventListener("DOMContentLoaded", () => {
```

File: src/index.js
Comment (lopopolo) [inline src/index.js]: I think `textContent` is the right property to set here.

```suggestion
  localeSeletor. textContent = isZhCN() ? 'English' : "中文";
```

Once #449 is merged, I think we'll want to replace these hard coded strings with strings loaded from the locale files.

**Review by lopopolo — COMMENTED** [review]
Left a couple of bits of feedback. @mattbasta I'd love your help here too if you can lend it.

I think I'd like to sequence things so this PR lands after #449.

File: src/index.js
Comment (mattbasta) [reply src/index.js]: The `textContent` change is good, but beware changing the locale names based on the current locale. If the UI is in Chinese and you only speak English, you would not want to find the menu item for English if it's written with Chinese characters.

File: src/index.js
Comment (mattbasta) [inline src/index.js]: Consider using `window.location.origin`, which includes the protocol and hostname together. It also does port, if you're running a local dev env

File: src/index.js
Comment (lopopolo) [reply src/index.js]: Is using flag emojis a way to solve this issue?

File: src/index.js
Comment (mattbasta) [reply src/index.js]: If this file gets compiled to `main.bundle.js`, replace `async` with `defer` in the HTML and you don't even need an event handler, just execute it immediately.

File: src/partials/nav.html
Comment (mattbasta) [inline src/partials/nav.html]: Add `role` with your JS; if JS fails to load, this will be a broken button for users with screen readers.

File: src/index.js
Comment (mattbasta) [inline src/index.js]: You can replace this whole function with `window.location.assign(getLocalizedPathname(window.location.pathname));`

File: src/index.js
Comment (mattbasta) [reply src/index.js]: Flags are even worse, because lots of countries have multiple very popular languages (Canada, India, China, etc.). In a language selector, write out the language names in the language spoken by the language being described.

Comment (lopopolo) [reply src/index.js]: OK, does that mean a dropdown like this in the navbar is more appropriate?

| 🗣 |
| -------|
| English |
| 中文 |

~This changes the selector to use _putonghua_ (Mandarin) rather than _zhongwen_ (Chinese nationality).~ (whoops I just checked with my wife. I was getting confused between _zhongwen_ and _zhongguoren_).

Or maybe like:

```
| ---------- |
| 🗣 English |
| ---------- | (hidden in dropdown)
|    中文    |
| ---------- |
```

Comment (mattbasta) [reply src/index.js]: Google has an icon for language that's free to use, iirc. That's pretty well established if you want to hide it in a drop-down

File: src/index.js
Comment (renny-ren) [reply src/index.js]: changed to defer.

File: src/partials/nav.html
Comment (renny-ren) [reply src/partials/nav.html]: Added

File: src/index.js
Comment (renny-ren) [reply src/index.js]: Cool

File: src/index.js
Comment (renny-ren) [reply src/index.js]: I think hide it in a dropdown is more appropriate for multi languages. It's easy to switch if there's only two languages for now.

File: src/partials/nav.html
Comment (lopopolo) [reply src/partials/nav.html]: Since we're adding this with JS, let's remove it from the HTML source:

```suggestion
```

**Review by lopopolo — CHANGES_REQUESTED** [review]
this looks mostly good to go, I think one tweak with the button in the HTML source.

**Comment by lopopolo** [comment]
@renny-ren you can fix the linter errors by running `npm run fmt`.

File: build.mjs
Comment (renny-ren) [inline build.mjs]: I can see this change is on artichoke:trunk already, but why it shows the diff here? it's weird.

Comment (lopopolo) [reply build.mjs]: I'm not sure what happened,  but I fixed up the branch in your fork.

**Comment by lopopolo** [comment]
Build is passing, I'll review this sometime in the next couple of days. Once it is approved though, I'm going to hold off merging until #449 lands.

**Comment by lopopolo** [comment]
Thanks for the PR @renny-ren. I'm going to close this in favor of https://github.com/artichoke/www.artichokeruby.org/pull/473.

**Pull request closed without merging**
